File: MVVMFrame/src/main/java/com/leonyr/lib/mvvm/act/AbActivity.java
package com.leonyr.lib.mvvm.act;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.Window;

/**
 * ==============================================================
 * Description: activity 基类
 * <p>
 * Created by leonyr on 2019-04-20.04.20
 * (C) Copyright LeonyR Corporation 2014 All Rights Reserved.
 * ==============================================================
 */
public abstract class AbActivity extends AppCompatActivity {

    protected String TAG;
    protected Context mCtx;
    private View rootView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        super.onCreate(savedInstanceState);
        mCtx = this;
        getLifecycle().addObserver(new ActivityObserver());

        setContentView(getLayoutResId());
        TAG = getClass().getSimpleName();

        initView();
    }

    @Override
    public void setContentView(int layoutResID) {
        rootView = View.inflate(this, layoutResID, null);
        super.setContentView(rootView);
    }

    public View getRootView() {
        return rootView;
    }

    @Override
    protected void onResume() {
        super.onResume();
        TAG = getClass().getSimpleName();
    }

    @Override
    protected void onDestroy() {
        mCtx = null;
        super.onDestroy();
    }

    /**
     * 获取layout id
     */
    @LayoutRes
    protected abstract int getLayoutResId();

    /**
     * 初始化各类view对象
     */
    protected abstract void initView();

}

File: MVVMFrame/src/main/java/com/leonyr/mvvm/act/AbActivity.kt
package com.leonyr.mvvm.act

import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.Window
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.Observer
import com.leonyr.mvvm.net.NetLoading


import com.leonyr.mvvm.vm.LViewModel

/**
 * ==============================================================
 * Description: activity 基类
 *
 *
 * Created by leonyr on 2019-04-20.04.20
 * (C) Copyright LeonyR Corporation 2014 All Rights Reserved.
 * ==============================================================
 */
abstract class AbActivity<VM : LViewModel> : AppCompatActivity() {

    lateinit var dialogLoading: NetLoading

    lateinit var TAG: String
    lateinit var mCtx: Context
    lateinit var rootView: View
        private set

    lateinit var vModel: VM

    /**
     * 获取layout id
     */
    @get:LayoutRes
    protected abstract val layoutResId: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        super.onCreate(savedInstanceState)
        mCtx = this
        dialogLoading = NetLoading(mCtx!!)
        lifecycle.addObserver(ActivityObserver())

        setContentView(layoutResId)
        TAG = javaClass.simpleName

        initView()
    }

    fun observerLoading() {
        vModel.showLoading.observe(this, Observer { show ->
            if (show) {
                dialogLoading.show()
            } else {
                dialogLoading.dismiss()
            }
        })
    }

    override fun setContentView(layoutResID: Int) {
        rootView = View.inflate(this, layoutResID, null)
        super.setContentView(rootView)
    }

    override fun onResume() {
        super.onResume()
        TAG = javaClass.simpleName

    }

    /**
     * 初始化各类view对象
     */
    protected abstract fun initView()

    /*override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)

        // Forward results to EasyPermissions
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this)
    }*/
}
